### DISCUSSION TITLE
v3.3.0

### DISCUSSION THREAD

**Discussion by krashish8** [pgRouting/pgrouting#2213] [Announcements]
pgRouting 3.3.0 Release Notes
-------------------------------------------------------------------------------

To see all issues & pull requests closed by this release see the [Git closed milestone for 3.3.0](https://github.com/pgRouting/pgrouting/issues?utf8=%E2%9C%93&q=milestone%3A%22Release%203.3.0%22) on Github.

**Issue fixes**

* [#2057](https://github.com/pgRouting/pgrouting/issues/2057): trspViaEdges columns in different order
* [#2087](https://github.com/pgRouting/pgrouting/issues/2087): pgr_extractVertices to proposed
* [#2201](https://github.com/pgRouting/pgrouting/issues/2201): pgr_depthFirstSearch to proposed
* [#2202](https://github.com/pgRouting/pgrouting/issues/2202): pgr_sequentialVertexColoring to proposed
* [#2203](https://github.com/pgRouting/pgrouting/issues/2203): pgr_dijkstraNear and pgr_dijkstraNearCost to proposed

**New experimental functions**

* Coloring

  * pgr_edgeColoring

**Experimental promoted to Proposed**

* Dijkstra

  * pgr_dijkstraNear

    * pgr_dijkstraNear(Combinations)
    * pgr_dijkstraNear(Many to Many)
    * pgr_dijkstraNear(Many to One)
    * pgr_dijkstraNear(One to Many)

  * pgr_dijkstraNearCost

    * pgr_dijkstraNearCost(Combinations)
    * pgr_dijkstraNearCost(Many to Many)
    * pgr_dijkstraNearCost(Many to One)
    * pgr_dijkstraNearCost(One to Many)

* Coloring

  * pgr_sequentialVertexColoring

* Topology

  * pgr_extractVertices

* Traversal

  * pgr_depthFirstSearch(Multiple vertices)
  * pgr_depthFirstSearch(Single vertex)

**Attachments**
File | Contents
| --- | --- |
| `doc-v3.3.0-en-es.tar.gz` | English and Spanish documentation. Redirection to English
| `doc-v3.3.0-en.tar.gz`|English documentation. Redirection to English
| `doc-v3.3.0-es.tar.gz`|Spanish documentation. Redirection to Spanish
| `pgrouting-3.3.0.tar.gz` | tar.gz of the release
| `pgrouting-3.3.0.zip`| zip of the release

<hr /><em>This discussion was created from the release <a href='https://github.com/pgRouting/pgrouting/releases/tag/v3.3.0'>v3.3.0</a>.</em>

**Comment by krashish8** [comment]
## What's Changed
* pumping up version to 3.3.0-dev by @cvvergara in https://github.com/pgRouting/pgrouting/pull/1890
* [build] remove accidentally added file by @cvvergara in https://github.com/pgRouting/pgrouting/pull/1932
* Upgrade action tester dev by @cvvergara in https://github.com/pgRouting/pgrouting/pull/1934
* Code style dev by @cvvergara in https://github.com/pgRouting/pgrouting/pull/1936
* fix for Issue1894 Column_info_t #1894  by @Lee-Pakkun in https://github.com/pgRouting/pgrouting/pull/1937
* Porting changes of main by @cvvergara in https://github.com/pgRouting/pgrouting/pull/1973
* [alpha_shape] removing directive PGSQL_VERSION > 95  by @stratospher in https://github.com/pgRouting/pgrouting/pull/1974
* [breadthFirstSearch] removing directive PGSQL_VERSION > 95 by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/1979
* [astar] removing directive PGSQL_VERSION > 95 by @rdotjain in https://github.com/pgRouting/pgrouting/pull/1975
* [bdAstar] removing directive PGSQL_VERSION > 95 by @raghavTinker in https://github.com/pgRouting/pgrouting/pull/1976
* [max_flow] removing directive PGSQL_VERSION > 95 by @king-11 in https://github.com/pgRouting/pgrouting/pull/1977
* [bellman_ford] removing directive PGSQL_VERSION > 95 by @king-11 in https://github.com/pgRouting/pgrouting/pull/1978
* fix for Issue1908 pgr_components_rt by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/1981
* [dijkstra] removing directive PGSQL_VERSION > 95 by @bkrit in https://github.com/pgRouting/pgrouting/pull/1980
* fix for Issue1895 contracted_rt by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/1982
* [spanningTree] removing directive PGSQL_VERSION > 95 by @king-11 in https://github.com/pgRouting/pgrouting/pull/1983
* [topologicalSort] removing directive PGSQL_VERSION > 95 by @king-11 in https://github.com/pgRouting/pgrouting/pull/1985
* [bdDijkstra] removing directive PGSQL_VERSION > 95 by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/1986
* [chinese] removing directive PGSQL_VERSION > 95 by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/1987
* [tsp] removing directive PGSQL_DIRECTIVE > 95 by @king-11 in https://github.com/pgRouting/pgrouting/pull/1988
* fix for Issue1896 Coordinate_t by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/1989
* [traversal] removing directive PGSQL_VERSION > 95 by @gauravagrwal in https://github.com/pgRouting/pgrouting/pull/1990
* [src/dagShortestPath] removing directive PGSQL_VERSION > 95 by @vicilanaya in https://github.com/pgRouting/pgrouting/pull/1993
* fix for Issue1897 Delauny_t  by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/1992
* fix for Issue1921 pgr_vertex_color_rt by @king-11 in https://github.com/pgRouting/pgrouting/pull/1996
* [src/trsp] removing directive PGSQL_VERSION > 95 by @vicilanaya in https://github.com/pgRouting/pgrouting/pull/1997
* [src/coloring] removing directive PGSQL_VERSION > 95 by @vicilanaya in https://github.com/pgRouting/pgrouting/pull/1999
* [src/mincut] removing directive PGSQL_VERSION > 94 by @a-gon in https://github.com/pgRouting/pgrouting/pull/1998
* [src/withPoints] removing directive PGSQL_VERSION > 95 by @vicilanaya in https://github.com/pgRouting/pgrouting/pull/2000
* [src/contraction] removing directive PGSQL_VERSION > 95 by @vicilanaya in https://github.com/pgRouting/pgrouting/pull/2002
* [src/dominator] removing directive PGSQL_VERSION > 95 by @vicilanaya in https://github.com/pgRouting/pgrouting/pull/2003
* [src/pickDeliver] removing directive PGSQL_VERSION > 95 by @a-gon in https://github.com/pgRouting/pgrouting/pull/2005
* fix for Issue1898 General_path_element_t by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/2007
* [src/components] removing directive PGSQL_VERSION > 95 by @vicilanaya in https://github.com/pgRouting/pgrouting/pull/2006
* [src/driving_distance] removing directive PGSQL_VERSION > 95 by @vicilanaya in https://github.com/pgRouting/pgrouting/pull/2004
* [src/transitiveClosure] removing directive PGSQL_VERSION > 95 by @vicilanaya in https://github.com/pgRouting/pgrouting/pull/2008
* fix for Issue1903 Matrix_cell_t by @cvvergara in https://github.com/pgRouting/pgrouting/pull/1931
* [allpairs] removing directive PGSQL_VERSION > 95 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/1967
* fix for 1912 adjustments for pgr_flow_t by @king-11 in https://github.com/pgRouting/pgrouting/pull/2009
* fix for 1913 pgr_ltdtree_rt.h by @king-11 in https://github.com/pgRouting/pgrouting/pull/2010
* fix 1923 pickDeliveryOrders_t.h by @king-11 in https://github.com/pgRouting/pgrouting/pull/2011
* fix for Issue1901 Line_graph_full_rt by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/2012
* fix for Issue1902 Line_graph_rt by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/2013
* Issue1959 by @Ri7Sh in https://github.com/pgRouting/pgrouting/pull/2014
* fix for Issue1904 pgr_basic_edge_t by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/2017
* [src/ksp] removing directive PGSQL_VERSION > 95 by @Ri7Sh in https://github.com/pgRouting/pgrouting/pull/2015
* [src/lineGraph] removing directive PGSQL_VERSION > 95 by @Ri7Sh in https://github.com/pgRouting/pgrouting/pull/2016
* Fixing Issue1905 pgr_bipartite_rt by @Ri7Sh in https://github.com/pgRouting/pgrouting/pull/2019
* Fixing Issue1920 on pgr_topologicalSort_t by @Ri7Sh in https://github.com/pgRouting/pgrouting/pull/2020
* fix for issue1906 pgr_boyer_t by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/2021
* fix for Issue1909 pgr_costFlow_t by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/2022
* fix for Issue1907 pgr_combination_t by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/2024
* Fix stylecheck by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2023
* fix for Issue1924 Vehicle_t by @ManasviGoyal in https://github.com/pgRouting/pgrouting/pull/2028
* fixed issue1916 Pgr_point_t by @ManasviGoyal in https://github.com/pgRouting/pgrouting/pull/2032
* fix for Issue1914 pgr_makeConnected_t by @nishkakotian in https://github.com/pgRouting/pgrouting/pull/2033
* fix for Issue1918 pgr_randomSpanningTree_t by @ManasviGoyal in https://github.com/pgRouting/pgrouting/pull/2039
* Issue1927 include/c_types/routes_t.h by @Ri7Sh in https://github.com/pgRouting/pgrouting/pull/2037
* [c_types][C/C++] fixing edge_t.h by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2040
* Making compilation faster  include/c_types/trsp_types.h by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2041
* Making compilation faster transitiveClosure_rt by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2042
* Making compilation faster pgr_prim_rt.h  by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2045
* Making compilation faster mst_rt.h by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2046
* Making compilation faster include/c_types/restriction_t.h by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2043
* Making compilation faster include/c_types/point_on_edge_t.h by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2044
* Making compilation faster pgr stoer wagner t by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2048
* Making compilation faster general vehicle orders t by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2049
* Adjust drivers in allpairs by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2052
* Adjust drivers in astar by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2055
* Adjust drivers in bdAstar by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2058
* [GSoC-2021] Experimental Function - pgr_edgeColoring  by @veenits123 in https://github.com/pgRouting/pgrouting/pull/2061
* Making compilation faster, chaging pgr_edge_xy_t.h by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2062
* Adjust drivers in bdDijkstra by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2064
* Adjust drivers in bellman_ford by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2066
* Adjust drivers in chinese by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2070
* Adjust drivers in breadthFirstSearch by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2068
* Adjust drivers in coloring by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2072
* Adjust drivers in components by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2074
* Adjust drivers in contraction by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2076
* Adjust drivers in dagShortestPath by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2078
* Adjust drivers in dijkstra by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2080
* Adjust drivers in dominator by @abhichou4 in https://github.com/pgRouting/pgrouting/pull/2082
* Fixing 2 issues for trsp by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2059
* [GSoC-2021] Updating pgr_edgeColoring Doc by @veenits123 in https://github.com/pgRouting/pgrouting/pull/2084
* Porting tsp fix dev by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2083
* Fixed Issue1899 for  geom_text_rt.h by @pranavajitnair in https://github.com/pgRouting/pgrouting/pull/2088
* Adjust drivers: max_flow by @samyakjain0112 in https://github.com/pgRouting/pgrouting/pull/2092
* Compilation on visual studio dev by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2095
* [CI][Windows] github Action to build in windows by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2097
* C types reorganization and cleanup by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2103
* Adjust drivers: topologicalSort by @harshitagrawal294 in https://github.com/pgRouting/pgrouting/pull/2105
* Adjust drivers: yen by @harshitagrawal294 in https://github.com/pgRouting/pgrouting/pull/2107
* Links to old pages update by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2098
* [contraction] fixing PLSQL by @shonnungar in https://github.com/pgRouting/pgrouting/pull/2144
* [alpha_shape] fixing PLSQL by @bkrit in https://github.com/pgRouting/pgrouting/pull/2145
* Add automatic locale update in GitHub actions by @krashish8 in https://github.com/pgRouting/pgrouting/pull/2146
* Run locale & doc actions only on the main repository by @krashish8 in https://github.com/pgRouting/pgrouting/pull/2147
* [breadthFirstSearch] Fixing PSQL by @h-murphy in https://github.com/pgRouting/pgrouting/pull/2148
* [astar] fixing PLSQL by @alisonrwu in https://github.com/pgRouting/pgrouting/pull/2150
* [driving_distance] fixing PLSQL by @shrija14 in https://github.com/pgRouting/pgrouting/pull/2152
* [mincut] fixing PLSQL by @jsteele13 in https://github.com/pgRouting/pgrouting/pull/2154
* [components] fixing PLSQL by @shrija14 in https://github.com/pgRouting/pgrouting/pull/2155
* [max_flow] fixing PLSQL by @jsteele13 in https://github.com/pgRouting/pgrouting/pull/2156
* [dominator] fixing PLSQL by @shrija14 in https://github.com/pgRouting/pgrouting/pull/2158
* [mincut] Adjust drivers by @alisonrwu in https://github.com/pgRouting/pgrouting/pull/2160
* [Dijkstra] fixing PLSQL #2135 by @RediIssac in https://github.com/pgRouting/pgrouting/pull/2153
* [ksp] fixing PLSQL by @shrija14 in https://github.com/pgRouting/pgrouting/pull/2163
* [chinese] fixing PLSQL by @alisonrwu in https://github.com/pgRouting/pgrouting/pull/2162
* [bdAstar] fixing PLSQL Issue# 2125 by @madrinathapa in https://github.com/pgRouting/pgrouting/pull/2164
* [allpairs] fixing PLSQL by @Motahareee in https://github.com/pgRouting/pgrouting/pull/2166
* [pickDeliver] fixing PLSQL #2142 by @madrinathapa in https://github.com/pgRouting/pgrouting/pull/2161
* [doc] pgr_version.rst Corrected typo by @thayeray in https://github.com/pgRouting/pgrouting/pull/2168
* Fix documentation references to results by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2169
* [tsp] adjusting drivers by @shrija14 in https://github.com/pgRouting/pgrouting/pull/2172
* [withPoints] adjusting drivers by @shrija14 in https://github.com/pgRouting/pgrouting/pull/2171
* [dagShortestPath] fixing PLSQL by @shrija14 in https://github.com/pgRouting/pgrouting/pull/2173
* [coloring] fixing PLSQL by @shrija14 in https://github.com/pgRouting/pgrouting/pull/2174
* [traversal] adjusting drivers by @shrija14 in https://github.com/pgRouting/pgrouting/pull/2175
* [transitiveClosure] adjusting drivers by @shrija14 in https://github.com/pgRouting/pgrouting/pull/2176
* [lineGraph] fixing PLSQL by @marinara28 in https://github.com/pgRouting/pgrouting/pull/2177
* [doc] fixed documentation typos/gramaar by @thayeray in https://github.com/pgRouting/pgrouting/pull/2170
* [bellman_ford] fixing PLSQL by @h-murphy in https://github.com/pgRouting/pgrouting/pull/2149
* [build][doc] pumping up shpinx version by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2179
* Adjust drivers by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2180
* Issue 2120 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2181
* Issue 2087 pgr_extractVertices by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2182
* Fixing codacy errors on pgtap directory by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2183
* [tools] Removing codacy errors on shell scripts by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2184
* Fix more codacy errors sql by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2185
* Fix codacy remaining errors by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2186
* Coday remaining errors fix by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2187
* Rhel7 fix dev by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2194
* Issue 2203 pg dijkstra near to proposed by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2205
* [depthFirstSearch] promote to proposed by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2206
* [pgr_sequentialVertexColoring] promoted to proposed by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2207
* Issue 2208 tasks before release by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2209
* [doc] fix some mistakes by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2210
* Translations to spanish nov 2021 by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2211
* Develop by @cvvergara in https://github.com/pgRouting/pgrouting/pull/2212

## New Contributors
* @Lee-Pakkun made their first contribution in https://github.com/pgRouting/pgrouting/pull/1937
* @stratospher made their first contribution in https://github.com/pgRouting/pgrouting/pull/1974
* @nishkakotian made their first contribution in https://github.com/pgRouting/pgrouting/pull/1979
* @rdotjain made their first contribution in https://github.com/pgRouting/pgrouting/pull/1975
* @raghavTinker made their first contribution in https://github.com/pgRouting/pgrouting/pull/1976
* @king-11 made their first contribution in https://github.com/pgRouting/pgrouting/pull/1977
* @bkrit made their first contribution in https://github.com/pgRouting/pgrouting/pull/1980
* @gauravagrwal made their first contribution in https://github.com/pgRouting/pgrouting/pull/1990
* @vicilanaya made their first contribution in https://github.com/pgRouting/pgrouting/pull/1993
* @a-gon made their first contribution in https://github.com/pgRouting/pgrouting/pull/1998
* @Ri7Sh made their first contribution in https://github.com/pgRouting/pgrouting/pull/2014
* @ManasviGoyal made their first contribution in https://github.com/pgRouting/pgrouting/pull/2028
* @abhichou4 made their first contribution in https://github.com/pgRouting/pgrouting/pull/2052
* @veenits123 made their first contribution in https://github.com/pgRouting/pgrouting/pull/2061
* @pranavajitnair made their first contribution in https://github.com/pgRouting/pgrouting/pull/2088
* @samyakjain0112 made their first contribution in https://github.com/pgRouting/pgrouting/pull/2092
* @harshitagrawal294 made their first contribution in https://github.com/pgRouting/pgrouting/pull/2105
* @shonnungar made their first contribution in https://github.com/pgRouting/pgrouting/pull/2144
* @h-murphy made their first contribution in https://github.com/pgRouting/pgrouting/pull/2148
* @alisonrwu made their first contribution in https://github.com/pgRouting/pgrouting/pull/2150
* @shrija14 made their first contribution in https://github.com/pgRouting/pgrouting/pull/2152
* @jsteele13 made their first contribution in https://github.com/pgRouting/pgrouting/pull/2154
* @RediIssac made their first contribution in https://github.com/pgRouting/pgrouting/pull/2153
* @madrinathapa made their first contribution in https://github.com/pgRouting/pgrouting/pull/2164
* @Motahareee made their first contribution in https://github.com/pgRouting/pgrouting/pull/2166
* @marinara28 made their first contribution in https://github.com/pgRouting/pgrouting/pull/2177

**Full Changelog**: https://github.com/pgRouting/pgrouting/compare/v3.2.2...v3.3.0

**Comment by dkastl** [comment]
... and I just realised now that "Discussions" are used ;-)
